### ISSUE TITLE
Unnecessary spaces

### ISSUE THREAD

**Issue by HendrikWismans** [VincentLanglet/Twig-CS-Fixer#141]
Hallo!

W're using this libary and it works fine! Thanks!

There are already some issues about spaces. But in my opnion not all double spaces that can removed are removed. Below are just some samples. I guess the're more cases.

### Sample code
```
{% extends                '@Admin/...' %}
{% set     test = 'test' %}
{% if    test is       not null %}{% endif %}
```
### Expected behavior

```
{% extends '@Admin/...' %}
{% set test = 'test' %}
{% if test is not null %}{% endif %}
```
### Current output after running twig-cs-fixer lint --fix
```
{% extends                '@Admin/...' %}
{% set     test = 'test' %}
{% if    test is       not null %}{% endif %}
```

**Comment by VincentLanglet** [comment]
Hi, thanks for the report.

I see two different issue here:

1) The `is.....not` issue.

I think Twig consider that `is not` is a single token. So does my library.
That's understandable since `foo is not bar` is not the same than `foo is (not bar)`.

`is not` is consider as a operator (Like , even if there is space inside.)

We can see `// An operator with a space can be any amount of whitespaces`  in the twig code, and regex are done this way:
![image](https://github.com/VincentLanglet/Twig-CS-Fixer/assets/9052536/0b947646-b556-4d8c-8413-7a8fb3fb3a02)

So I assume the same issue can exist with others operators like
```
{{ 'array'  starts    with  'a' }}
```

So we'll need to basically edit the value of the operator to replace every multiple space to a single space.
I dunno if this can have bad impacts when you have custom operators with spaces...

2) The `extends`/`set`/`if` issue.

These are `BLOCK_TAG_TYPE` is my project.

There is a `DelimiterSpacingSniff` for BLOCK_START_TYPE and BLOCK_END_TYPE because this is in the standard:
https://twig.symfony.com/doc/3.x/coding_standards.html
> Put one (and only one) space after the start of a delimiter ({{, {%, and {#) and before the end of a delimiter (}}, %}, and #}):

There is nothing about the block tag type, but this is a good idea.

**Comment by VincentLanglet** [comment]
https://github.com/VincentLanglet/Twig-CS-Fixer/pull/143 should already solve half of your problems

**Comment by VincentLanglet** [comment]
And https://github.com/VincentLanglet/Twig-CS-Fixer/pull/144 you do the other part.

Can you try dev-main @HendrikWismans ?

**Comment by HendrikWismans** [comment]
Thanks! works fine!

**Comment by VincentLanglet** [comment]
Released in 1.6.0
